pkg/pagerduty/pagerduty.go

```diff
@@ -186,6 +190,52 @@ func (c Client) GetAlerts(incidentID string) ([]Alert, error) {
 	return res, nil
 }
 
+// CreateNewAlert triggers an alert using the Deadmanssnitch integration for the given service.
+// If the provided service does not have a DMS integration, an error is returned
+func (c Client) CreateNewAlert(description string, details interface{}, serviceID string) error {
```

Comment:
Does details need to be an `interface{}`?

We seem to be using this struct as input whenever we call this in this PR, so I think we should define this specific struct and enforce the type on `details`

```go
struct {
		ClusterID  string `json:"Cluster ID"`
		Error      string `json:"Error"`
		Resolution string `json:"Resolution"`
		SOP        string `json:"SOP"`
	}
```

Reply:
At the moment, the `details` argument just reflects the `details` field for an [Event object in go-pagerduty](https://pkg.go.dev/github.com/pagerduty/go-pagerduty#Event), which is an `interface{}`. We could create a more structured-structure, but personally I like the flexibility it gives callers in defining only the fields relevant to the alert being created

Reply:
Thanks, mirroring upstream makes sense to me.